core/file/printf_spec.rb

```diff
@@ -0,0 +1,19 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+require File.expand_path('../../kernel/shared/sprintf', __FILE__)
+
+describe "File#printf" do
```

Comment:
It's `IO#printf`:
```ruby
[1] pry(main)> File.new("README.md").method :printf
=> #<Method: File(IO)#printf>
```

Reply:
@eregon Got it

I checked owner for StringIO and have found it is `IO::writable` in Ruby 2.2 and `IO::generic_writable` in Ruby 2.3/2.4. It looks like an implementation detail and I wonder if I should test `printf` for both `IO::writable` and `IO::generic_writable`

```ruby
StringIO.new.method(:printf).owner
=> IO::writable
```

Reply:
Interesting.
Those modules are definitely internal, even though they show up in `ancestors`:
```ruby
> StringIO.ancestors
=> [StringIO,
 IO::generic_writable,
 IO::generic_readable,
 Enumerable,
 Data,
 Object,
 PP::ObjectMixin,
 Kernel,
 BasicObject]
```

So I think it's best to share specs as possible and just have the specs in `StringIO#printf`, since modules above are internal.

Reply:
Could you make this a spec for IO#printf?

Reply:
@eregon do you propose to test only IO#printf or share the specs for ancestors too (File/StringIO)?

Reply:
@eregon ping